pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/DeleteAPIHybridClusterIntegrationTest.java

```diff
@@ -0,0 +1,247 @@
+        forDeleteAllSegmentsWithTypeWithGetAPI(TABLE_NAME, "realtime"));
+    Thread.sleep(3000);
+
+    String postDeleteSegmentList = sendGetRequest(
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forListAllSegments(TABLE_NAME));
+    Assert.assertEquals(getSegmentsFromJson(postDeleteSegmentList, "realtime").size(), 1);
+    Assert.assertNotSame(getSegmentsFromJson(postDeleteSegmentList, "offline").size(), 0);
+
+    repushRealtimeSegments();
+  }
+
+  @Test
+  public void deleteAllOfflineSegmentsFromGetAPI() throws Exception{
+    sendGetRequest(ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).
+        forDeleteAllSegmentsWithTypeWithGetAPI(TABLE_NAME, "offline"));
+    Thread.sleep(3000);
+
+    String postDeleteSegmentList = sendGetRequest(
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forListAllSegments(TABLE_NAME));
+    Assert.assertEquals(getSegmentsFromJson(postDeleteSegmentList, "offline"), null);
+    Assert.assertNotSame(getSegmentsFromJson(postDeleteSegmentList, "realtime").size(), 0);
+
+    repushOfflineSegments();
+  }
+
+  @Override
+  public void testHardcodedQuerySet() {}
+
+  @Test
+  public void deleteOfflineSegmentFromGetAPI() throws Exception{
+    String segmentList = sendGetRequest(
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forListAllSegments(TABLE_NAME));
+    JSONArray offlineSegmentsList = getSegmentsFromJson(segmentList, "offline");
+
+    sendGetRequest(ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).
+        forDeleteSegmentWithGetAPI(TABLE_NAME, offlineSegmentsList.get(0).toString(), "offline"));
+    Thread.sleep(5000);
+
+    String postDeleteSegmentList = sendGetRequest(
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forListAllSegments(TABLE_NAME));
+    JSONArray offlineSegmentsListReturn = getSegmentsFromJson(postDeleteSegmentList, "offline");
+    offlineSegmentsList.remove(0);
+    Assert.assertEquals(offlineSegmentsListReturn, offlineSegmentsList);
+  }
+
+  @Override
+  public void testGeneratedQueriesWithMultiValues() {}
+
+  @Test
+  public void deleteRealtimeSegmentFromGetAPI() throws Exception{
+    String segmentList = sendGetRequest(
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forListAllSegments(TABLE_NAME));
+    JSONArray realtimeSegmentsList = getSegmentsFromJson(segmentList, "realtime");
+
+    sendGetRequest(ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).
+        forDeleteSegmentWithGetAPI(TABLE_NAME, realtimeSegmentsList.get(0).toString(), "realtime"));
+    Thread.sleep(3000);
+
+    String postDeleteSegmentList = sendGetRequest(
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forListAllSegments(TABLE_NAME));
+    JSONArray realtimeSegmentsListReturn = getSegmentsFromJson(postDeleteSegmentList, "realtime");
+    realtimeSegmentsList.remove(0);
```

Comment:
do not assume the order will be same. Instead, make sure that the name of the segment does not exist in the new output